gematria/datasets/basic_block_utils.cc

```diff
@@ -123,4 +123,53 @@ std::vector<unsigned> getUsedRegisters(
   return UsedRegistersList;
 }
 
+unsigned getSuperRegisterAllClasses(unsigned OriginalRegister,
+                                    const MCRegisterInfo &RegisterInfo) {
+  unsigned SuperRegister = OriginalRegister;
+  for (MCPhysReg CurrentSuperRegister :
+       RegisterInfo.superregs_inclusive(OriginalRegister))
+    SuperRegister = CurrentSuperRegister;
```

Comment:
We should use the same check as in https://github.com/google/gematria/blob/ad44397422734b3e43765821b11842edc8c003e3/gematria/datasets/basic_block_utils.cc#L39 to make sure we're always moving towards the largest super register.